### PR TITLE
Improved build

### PR DESCRIPTION
**Pull request by MaceM8**

### CODE REVIEW

File: apps/storybook/.storybook/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		return rule;
+	});
+	// /\.(svg|ico|jpg|jpeg|png|gif|eot|otf|webp|ttf|woff|woff2|cur|ani)(\?.*)?$/
+	config.module.rules = config.module.rules.filter(
+		rule =>
+			String(rule.test) !==
+			String(/\.(svg|ico|jpg|jpeg|png|gif|eot|otf|webp|ttf|woff|woff2|cur|ani)(\?.*)?$/)
+	);
+
+	const merged = merge(rootConfig, config);
```
Comment (tommmyy) [inline apps/storybook/.storybook/webpack.config.js]: why is this needed? if there are some common functionalities between root config (scss loader probably?) and this one then use webpack-blocks

File: apps/storybook/package.json [hunk 2]
```diff
@@ -4,20 +4,30 @@
...
+		"@storybook/addon-actions": "5.3.3",
+		"@storybook/addon-info": "5.3.3",
+		"@storybook/addon-knobs": "5.3.3",
+		"@storybook/addon-options": "5.3.3",
+		"@storybook/react": "5.3.3",
+		"@storybook/theming": "5.3.3",
+		"autoprefixer": "^9.7.4",
+		"babel-loader": "^8.0.6",
+		"husky": "^4.0.9",
+		"lundium": "^0.0.1",
```
Comment (tommmyy) [inline apps/storybook/package.json]: This is a dependency

File: apps/storybook/package.json [hunk 3]
```diff
@@ -4,20 +4,30 @@
...
 		"build": "build-storybook -o ../../build/storybook -c .storybook",
 		"start": "start-storybook -p 8001 -c .storybook"
+	},
+	"dependencies": {
+		"react": "^16.12.0",
+		"react-dom": "^16.12.0"
+	},
+	"devDependencies": {
+		"@babel/core": "^7.8.3",
+		"@babel/preset-env": "^7.8.3",
```
Comment (tommmyy) [inline apps/storybook/package.json]: not needed, we use babel-preset-react-union (remember our discussion about `rootMode: "upward"`

File: package.json [hunk 4]
```diff
@@ -23,11 +28,13 @@
...
 		"test": "jest",
 		"storybook": "yarn workspace @lundium/storybook",
-		"build:storybook": "yarn storybook build"
+		"create-stats": "webpack --profile --json > stats.json",
+		"analyze": "yarn create-stats && webpack-bundle-analyzer ./stats.json"
 	},
 	"devDependencies": {
 		"@babel/core": "^7.8.0",
 		"@babel/preset-env": "^7.8.2",
+		"@babel/preset-es2015": "^7.0.0-beta.53",
```
Comment (tommmyy) [inline package.json]: we do not use those presets. we use `babel-preset-react-union`

Comment (tommmyy) [reply package.json]: see babel.config.js

File: package.json [hunk 5]
```diff
@@ -44,10 +51,13 @@
 		"eslint-plugin-import": "^2.20.0",
 		"eslint-plugin-react": "^7.17.0",
 		"eslint-plugin-react-hooks": "^2.3.0",
+		"fs-extra": "^8.1.0",
+		"glob": "^7.1.6",
 		"husky": "^4.0.6",
 		"jest": "^24.9.0",
 		"lerna": "^3.20.2",
 		"lint-staged": "^9.5.0",
+		"material-ui": "^0.20.2",
```
Comment (tommmyy) [inline package.json]: what? 😸

File: packages/lundium/package.json [hunk 6]
```diff
@@ -1,9 +1,10 @@
 {
-	"name": "@lundium/components",
+	"name": "lundium",
 	"version": "0.0.1",
 	"license": "UNLICENSED",
 	"description": "Components",
-	"main": "./src/index.js",
+	"main": "./dist",
+	"sideEffects": false,
```
Comment (tommmyy) [inline packages/lundium/package.json:7]: are you sure?

File: packages/lundium/package.json [hunk 7]
```diff
@@ -1,9 +1,10 @@
...
 	"scripts": {},
```
Comment (tommmyy) [inline packages/lundium/package.json]: you can remove it

File: apps/storybook/.storybook/webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+const path = require('path');
+const { lstatSync, readdirSync } = require('fs');
+const merge = require('webpack-merge');
+
+const rootConfig = require('../../../webpack.config');
```
Comment (tommmyy) [inline apps/storybook/.storybook/webpack.config.js]: you removed this file or not?

File: tools/buildLundiumPackage.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+const entries = directoryPackages.reduce(
+	(entries, newEntry) => ({
+		...entries,
+		[newEntry]: `${relativeSrcPath}/${newEntry}/index.js`,
+	}),
+	{}
+);
+
+const compiler = webpack({
```
Comment (tommmyy) [inline tools/buildLundiumPackage.js]: please load the config from a file:
```
const webpackConfig = require("./webpack.config");

const compiler = webpack(compiler)
```
```

**Review by tommmyy — CHANGES_REQUESTED** [review]

File: tools/buildLundiumPackage.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+	output: {
+		filename: '[name].js',
+		path: path.resolve(`${lundiumPackagePath}/dist`),
+	},
+	module: {
+		rules: [
+			{
+				test: /\.js$/,
+				exclude: /node_modules/,
+				loader: 'babel-loader',
```
Comment (tommmyy) [inline tools/buildLundiumPackage.js]: missing `options: { rootMode: "upward" }`

**Comment by tommmyy** [comment]
@MaceM8 please remove:

```
'@babel/preset-env',
'@babel/preset-react',
```

from `babel.config.js` too. If you have issues with babel, I can look into that.

File: packages/lundium/package.json (hunk 6 above)
Comment (MaceM8) [reply packages/lundium/package.json:7]: Since there will be only UI components in this library, there shouldn't be any side-effects right? 
But maybe I'm missing something

File: tools/buildLundiumPackage.js (hunk 10 above)
Comment (MaceM8) [reply tools/buildLundiumPackage.js]: I've recreated it a bit, so this will be root config, since it should be primary build of this project

File: packages/lundium/package.json (hunk 6 above)
Comment (tommmyy) [reply packages/lundium/package.json:7]: there are... there will e.g. manipulation of the DOM
